### DISCUSSION TITLE
Compiling CP-editor source code

### DISCUSSION THREAD

**Discussion by hiensumi** [cpeditor/cpeditor#1258] [Q&A] [answered]
How can I compile the clone of CP-editor?

**Comment by ouuan** [comment, marked as answer]
https://cpeditor.org/docs/installation/build-from-source/

**Reply by hiensumi** [reply]
![image](https://github.com/cpeditor/cpeditor/assets/47080203/bd642332-721c-4941-b273-732fdcdd86d9)
I searched for the error on google but unable to fix it. Can you help me?

**Comment by ouuan** [comment]
The current master branch is harder to build. You need to install extra-cmake-modules and ksyntax-highlighting first. You can install them via the package manager or build from source. 

https://github.com/cpeditor/cpeditor/blob/4cd8904dc83aa8ccdc438d66ca0ec084ea8bd4f3/.github/workflows/build_test.yml#L58-L63

Or you can build an older version, which does not require ECM and KSH.
